hi-ml/src/health_ml/deep_learning_config.py

```diff
@@ -0,0 +1,451 @@
+    output_to: str = param.String(default="",
+                                  doc="If provided, the run outputs will be written to the given folder. If not "
+                                      "provided, outputs will go into a subfolder of the project root folder.")
+    file_system_config: ExperimentFolderHandler = param.ClassSelector(default=ExperimentFolderHandler(),
+                                                                      class_=ExperimentFolderHandler,
+                                                                      instantiate=False,
+                                                                      doc="File system related configs")
+    _model_name: str = param.String("", doc="The human readable name of the model (for example, Liver). This is "
+                                            "usually set from the class name.")
+
+    @property
+    def model_name(self) -> str:
+        """
+        Gets the human readable name of the model (e.g., Liver). This is usually set from the class name.
+        :return: A model name as a string.
+        """
+        return self._model_name
+
+    def set_output_to(self, output_to: PathOrString) -> None:
+        """
+        Adjusts the file system settings in the present object such that all outputs are written to the given folder.
+        :param output_to: The absolute path to a folder that should contain the outputs.
+        """
+        if isinstance(output_to, Path):
+            output_to = str(output_to)
+        self.output_to = output_to
+        self.create_filesystem()
+
+    def create_filesystem(self, project_root: Path = fixed_paths.repository_root_directory()) -> None:
+        """
+        Creates new file system settings (outputs folder, logs folder) based on the information stored in the
+        present object. If any of the folders do not yet exist, they are created.
+        :param project_root: The root folder for the codebase that triggers the training run.
+        """
+        self.file_system_config = ExperimentFolderHandler.create(
+            project_root=project_root,
+            model_name=self.model_name,
+            is_offline_run=not is_running_in_azure_ml(RUN_CONTEXT),
+            output_to=self.output_to
+        )
+
+    @property
+    def outputs_folder(self) -> Path:
+        """Gets the full path in which the model outputs should be stored."""
+        return self.file_system_config.outputs_folder
+
+    @property
+    def logs_folder(self) -> Path:
+        """Gets the full path in which the model logs should be stored."""
+        return self.file_system_config.logs_folder
+
+    @property
+    def checkpoint_folder(self) -> Path:
+        """Gets the full path in which the model checkpoints should be stored during training."""
+        return self.outputs_folder / CHECKPOINT_FOLDER
+
+
+class OptimizerParams(param.Parameterized):
+    l_rate: float = param.Number(1e-4, doc="The initial learning rate", bounds=(0, None))
```

Comment:
we have tons of schedulers here, prune back to the essential ones? Not sure how many we actually use in InnerEye?